src/Collection/Collection.router.ts

```diff
@@ -0,0 +1,92 @@
+import { server } from 'decentraland-server'
+import Ajv from 'ajv'
+
+import { Router } from '../common/Router'
+import { HTTPError, STATUS_CODES } from '../common/HTTPError'
+import { withAuthentication, withModelExists, AuthRequest } from '../middleware'
+import { withModelAuthorization } from '../middleware/authorization'
+import { Ownable } from '../Ownable'
+import { Collection } from '../Collection'
+import { collectionSchema } from './Collection.types'
+
+const ajv = new Ajv()
+
+export class CollectionRouter extends Router {
+  mount() {
+    const withCollectionExists = withModelExists(Collection, 'id')
+    const withCollectionAuthorization = withModelAuthorization(Collection)
+
+    /**
+     * Returns the collections for a user
+     */
+    this.router.get(
+      '/collections',
+      withAuthentication,
+      server.handleRequest(this.getCollections)
+    )
+
+    /**
+     * Returns a collection for a user
+     */
+    this.router.get(
+      '/collections/:id',
+      withAuthentication,
+      withCollectionExists,
+      withCollectionAuthorization,
+      server.handleRequest(this.getCollection)
+    )
+
+    /**
+     * Upserts the collection
+     */
+    this.router.put(
+      '/collections/:id',
+      withAuthentication,
+      withCollectionExists,
+      withCollectionAuthorization,
+      server.handleRequest(this.upsertCollection)
+    )
+  }
+
+  async getCollections(req: AuthRequest) {
+    const eth_address = req.auth.ethAddress
+    return Collection.find({ eth_address })
+  }
+
+  async getCollection(req: AuthRequest) {
+    const id = server.extractFromReq(req, 'id')
+    const eth_address = req.auth.ethAddress
+    return Collection.findOne({ id, eth_address })
```

Comment:
Is `eth_address` useful here?

Reply:
I thought about not having it here to be honest, but I left it as a check. But now that I think about it that's being handled on the middleware so no need. I'll remove it !